server/src/main/java/org/opensearch/index/translog/transfer/FileTransferException.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.translog.transfer;
+
+import org.opensearch.index.translog.transfer.FileSnapshot.TransferFileSnapshot;
+
+/**
+ * Exception when a single file transfer encounters a failure
+ *
+ * @opensearch.internal
+ */
+public class FileTransferException extends RuntimeException {
+
+    private final TransferFileSnapshot fileSnapshot;
+
+    public FileTransferException(TransferFileSnapshot fileSnapshot, Throwable cause) {
+        super(cause);
+        this.fileSnapshot = fileSnapshot;
+    }
+
+    public FileTransferException(TransferFileSnapshot fileSnapshot, String message, Throwable cause) {
```

Comment:
Is this constructor being used? I only see one instance of this exception being created.

Reply:
Will be used in later PRs will remove them for now